### DISCUSSION TITLE
How do I use the Nightly Rust Formatter with Auto Format?

### DISCUSSION THREAD

**Discussion by ShiromMakkad** [helix-editor/helix#11769] [Q&A]
My languages.toml:

```
[[language]]
name = "rust"
formatter = { command = "rustfmt" , args = ["+nightly"] }
auto-format = true

[language-server.rust-analyzer]
config = { cargo = { features = "all" } }
```

But when I write, nothing happens. I tried using `log-open` and searching for "rustfmt" but it didn't turn up any interesting log lines. Removing the `formatter` line makes the autoformat work properly, but just with the normal formatter. Are there some default args that helix uses? Or maybe helix invokes `cargo` to do formatting?

**Comment by Mirko-von-Leipzig** [comment]
I'm not sure what's going wrong on your end, but the following works as expected for me: 

```
[[language]]
name="rust"
formatter = { command = "rustfmt" , args = ["+nightly"] }
```

You could try remove the extra RA configuration to see if that's an issue somehow. Alternatively it might be that the `languages.toml` isn't being used.
